### ISSUE TITLE
File uploading functionality

### ISSUE THREAD

**Issue by liza-veis** [BinaryStudioAcademy/bsa-2023-calmpal#70]
[AWS S3](https://aws.amazon.com/s3/)

- [x] create files table according to db schema
- [x] files libs/package (package for file uploading to aws)
- [x] files package
  - [x] files model
  - [x] files repository
  - [x] files service
  - [x] files controller 
- [x] files fastify plugin
  - [x] file extension: jpeg/png
  - [x] file size: < 10Mb

**Comment by Nanaier** [comment]
@Liza-Veis hey, a quick question - is there already an AWS account to use or should I create one in my name + During the creation of the migration file for the files table, should I update the meditation_topics and user_details tables with the image_id/avatar_id columns?

**Comment by Nanaier** [comment]
@Liza-Veis quick question - How will we retrieve the file from AWS? Using presigned URL? or is there another way to do that? Cause for now there are access issues with file URL.

**Comment by v0ldemar01** [comment]
@Nanaier Absolutely. So we are storing link to s3 and for accessing the most truly way as a part of best practices is Presigned URL

**Comment by k-mlch** [comment]
Found issues:

**1. When the user tries to add a file > than 10MB** 

Expected Result: 400 Bad Request 
Actual Result: 500 Server Error returned

**2. When the user adds a file in any other format than .png or .jpg**

Expected Result: 400 Bad Request
Actual Result: 500 Server Error returned

**3. When the user doesn't add a file key to the request or adds empty file key**

Expected Result: 400 Bad Request
Actual Result: 500 Server Error returned

**Note**: Please update swagger documentation with this info too

**Comment by oksana-mlynska** [comment]
Also, the request body for the 201 must be updated. We need to return url, content-type and id, without message and make the appropriate change in Swagger too.


![Image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/140756353/a2ead043-c31c-4105-a3da-24732c561e7a)

**Comment by oksana-mlynska** [comment]
@Nanaier When I upload a valid file, I get in response not the ID of the authenticated user, but some other number and it changes each time I send the request, although token is not changed.
Or is it an ID of the uploaded file, not a user?

UPD: already understood that it's a file's ID

**Comment by oksana-mlynska** [comment]
@Nanaier please add to swagger documentation this response and also the response for > than 10MB file

![Image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/140756353/046594e0-7aa8-4808-bbec-85df1bd8f6a0)

**Comment by Nanaier** [comment]
@oksana-mlynska could you create a new ticket for that or should I just add a new fix to this ticket?

**Comment by oksana-mlynska** [comment]
@Nanaier could you please add a fix to this ticket? I'll put it in 'to do' column. You can take it when you have time.
